datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -883,6 +899,10 @@ pub fn array_append(args: &[ArrayRef]) -> Result<ArrayRef> {
 
 /// Array_sort SQL function
 pub fn array_sort(args: &[ArrayRef]) -> Result<ArrayRef> {
+    if args.is_empty() || args.len() > 3 {
```

Comment:
I think we need to make it more usable in future for all built in functions. To provide the exact available signatures, to let user quickly find what is his mistake

Reply:
Does that mean we need to check every kind of input in all the array function here?

Reply:
> I think we need to make it more usable in future for all built in functions. To provide the exact available signatures, to let user quickly find what is his mistake

@comphead 
But we will also handle this while checking signature of array. Why do we need to check it here again? Is there any usage that skip the signature checking but jump to this call directly? If yes, we need to find a way to reuse the checking because the coercion is not supported here currently, length checking may not be enough.

Reply:
Checking input arguments validity in public methods is good practice imho.

But the comment above was mostly to thinking on having a generic approach to return a friendly message to the user about what is wrong and what is the next step. Now we usually saying the input arguments has to be 1, 2, 3 arguments but it is mostly meaningless, we can probably improve it with more detailed message

Reply:
Sorry for the late reply. I agree with the message that more specific information would be better.